cub/cub/agent/agent_histogram.cuh

```diff
@@ -225,9 +225,9 @@ struct AgentHistogram
   // Wrap the native input pointer with CacheModifiedInputIterator
   // or directly use the supplied input iterator type
   using WrappedSampleIteratorT =
-    cub::detail::conditional_t<std::is_pointer<SampleIteratorT>::value,
-                               CacheModifiedInputIterator<LOAD_MODIFIER, SampleT, OffsetT>,
-                               SampleIteratorT>;
+    ::cuda::std::__conditional_t<std::is_pointer<SampleIteratorT>::value,
```

Comment:
Suggestion: There is a potential compile time improvement here.

`conditional_t` requies both alternatives to be instantiated. However, we have an alternative `::cuda::std::_If` with the same interface that only instantiates the taken alternative.

In most cases this does not make a difference but here we  could avoid instantiation of `CacheModifiedInputIterator<LOAD_MODIFIER, SampleT, OffsetT>`

Reply:
The reason `::cuda::std::_If` does not require instantiation is because the If and Else types only appear in alias declarations and never occur in class template parameters, right?